File: MOJ.ProductManagement.Application/DTOs/Product/ProductDto.cs
namespace MOJ.ProductManagement.Application.DTOs.Product
{
    public record ProductDto
    {
        public int Id { get; init; }
        public string Name { get; init; }
        public int QuantityPerUnitId { get; private set; }
        public int ReorderLevel { get; init; }
        public int SupplierId { get; init; }
        public decimal UnitPrice { get; init; }
        public int UnitsInStock { get; init; }
        public int UnitsOnOrder { get; init; }

        public string SupplierName { get; init; }
        public string QuantityPerUnitName { get; private set; }
    }
}

File: MOJ.ProductManagement.Web.Client.Angular/src/app/models/ProductDto.ts
export interface ProductDto {
    id: number;
    name: string;
    quantityPerUnitId: number;
    reorderLevel: number;
    supplierId: number;
    unitPrice: number;
    unitsInStock: number;
    unitsOnOrder: number;

    supplierName: string;
    quantityPerUnitName: string;
}